js/glpi_install.js

```diff
@@ -0,0 +1,85 @@
+ *
+ * @copyright 2015-2024 Teclib' and contributors.
+ * @licence   https://www.gnu.org/licenses/gpl-3.0.html
+ *
+ * ---------------------------------------------------------------------
+ *
+ * LICENSE
+ *
+ * This file is part of GLPI.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * ---------------------------------------------------------------------
+ */
+(() => {
+    function message(message_list_element, text) {
+        const alert = document.createElement('p');
+        alert.innerHTML = text;
+        message_list_element.appendChild(alert);
+    }
+
+    async function start_database_install(dom_element, progress_key)
+    {
+        if (!dom_element) {
+            throw new Error('No DOM element provided to start database install.');
+        }
+        const messages_container = document.getElementById('glpi_install_messages_container');
+        const success_element = document.getElementById('glpi_install_success');
+
+        const message_list_element = document.createElement('div');
+
+        success_element.querySelector('button').setAttribute('disabled', 'disabled');
+
+        const create_progress_bar = window.create_progress_bar;
+
+        if (typeof create_progress_bar === 'undefined') {
+            throw new Error('Function "create_progress_bar" is not defined. Did you load the associated JS file correctly?');
+        }
+
+        const progress = create_progress_bar({
+            key: progress_key,
+            container: messages_container,
+            error_callback: (error) => message(message_list_element, `Progress error:\n${error}`),
+        });
+
+        messages_container.appendChild(message_list_element);
+
+        setTimeout(() => {
+            progress.start();
+        }, 1500);
+
+        try {
+            const res = await fetch("/install/database_setup/start_db_inserts", {method: 'POST'});
+            const text = await res.text();
+            if (text && text.trim().length) {
+                message(message_list_element, `Error:\n${text}`);
+                progress.error();
+            } else {
+                success_element.querySelector('button').removeAttribute('disabled');
+            }
+        } catch (err) {
+            message(message_list_element, `Database install error:\n${err.message||err.toString()}`);
+            progress.error();
+        }
```

Comment:
As already said, you cannot thrust the result of this endpoint. Indeed, this request will probably still timeouts if there is a proxy between the client and the PHP server (e.g. nginx + php-fpm), so you cannot rely on its response.

The success/error state should be reported by the `/progress/check/`  endpoint and should trigger a call to the  `error_callback`  in case of a failure and to a `success_callback` in case of a success.

You could add a `updated_at` property to be able to consider that a lack of update in the latest XXX seconds corresponds to a timeout.

```suggestion
        try {
            const res = await fetch("/install/database_setup/start_db_inserts", {method: 'POST'});
        } catch (err) {
        }
```